### PR TITLE
feat(quick-upgrade): Add silver & silver needed

### PR DESCRIPTION
**Pull request by phated**
Found myself wanting to know exactly how much silver a planet has and how much for each rank. Needed to open the upgrade pane to find it, so figured I'd just add it here.

### CODE REVIEW

**Comment by jacobrosenthal** [comment]
Do you want to do any special casing for cant upgrade?
Currently just says

Silver: 0
Required:00000

**Comment by phated** [comment]
> Do you want to do any special casing for cant upgrade?
Currently just says
>
> Silver: 0
> Required:00000

Is this a lvl 0 planet? For non-planets it says "No planet selected or can't upgrade"

**Comment by jacobrosenthal** [comment]
Yeah my homeplanet

**Review by Stx69 — APPROVED** [review]

**Review by Stx69 — APPROVED** [review]

**Comment by andrewhong5297** [comment]
not sure if there is a way to check for a pending upgrade transaction, but otherwise it all seems to work!

File: content/productivity/quick-upgrade/plugin.js [hunk 1]
```diff
@@ -150,22 +189,42 @@ function UpgradeAllPlanets() {
...
+  let [selectedPlanet, setSelectedPlanet] = useState(() => {
+    const planet = ui.getSelectedPlanet();
+    if (!planet) {
+      return undefined;
     }
-    window.addEventListener('click', onClick);
-
-    return () => {
-      window.removeEventListener('click', onClick);
+    if (planet.planetType === PlanetType.PLANET) {
```
Comment (jacobrosenthal) [inline content/productivity/quick-upgrade/plugin.js:197]:
```suggestion
    if (planet.planetType === PlanetType.PLANET && planet.planetLevel > 0) {
```

File: content/productivity/quick-upgrade/plugin.js [hunk 2]
```diff
@@ -150,22 +189,42 @@ function UpgradeAllPlanets() {
...
+
+  useEffect(() => {
+    const sub = ui.selectedPlanetId$.subscribe(planetId => {
+      setSelectedPlanet(() => {
+        const planet = df.getPlanetWithId(planetId);
+        if (!planet) {
+          return undefined;
+        }
+
+        if (planet.planetType === PlanetType.PLANET) {
```
Comment (jacobrosenthal) [inline content/productivity/quick-upgrade/plugin.js:212]:
```suggestion
        if (planet.planetType === PlanetType.PLANET && planet.planetLevel > 0) {
```

Comment (phated) [reply content/productivity/quick-upgrade/plugin.js:212]: Thanks

**Comment by phated** [comment]
> not sure if there is a way to check for a pending upgrade transaction

There is, however, you can actually mash an upgrade if you have enough silver for 2 and it'll upgrade twice.

**Review by jacobrosenthal — APPROVED** [review]

**Pull request merged**